core/DbHelper.php

```diff
@@ -204,6 +204,57 @@ public static function tableHasIndex($table, $indexName)
         return !empty($result);
     }
 
+    /**
+     * Returns the default database charset to use
+     *
+     * Returns utf8mb4 if supported, with fallback to utf8
+     *
+     * @return string
+     * @throws Tracker\Db\DbException
+     */
+    public static function getDefaultCharset()
+    {
+        $result = Db::get()->fetchRow("SHOW CHARACTER SET LIKE 'utf8mb4'");
+
+        if (empty($result)) {
+            return 'utf8'; // charset not available
+        }
+
+        $result = Db::get()->fetchRow("SHOW VARIABLES LIKE 'character_set_database'");
+
+        if (!empty($result) && $result['Value'] === 'utf8mb4') {
+            return 'utf8mb4'; // database has utf8mb4 charset, so assume it can be used
+        }
+
+        $result = Db::get()->fetchRow("SHOW VARIABLES LIKE 'innodb_file_per_table'");
+
+        if (empty($result) || $result['Value'] !== 'ON') {
+            return 'utf8'; // innodb_file_per_table is required for utf8mb4
+        }
+
+        return 'utf8mb4';
+    }
+
+    /**
+     * Returns sql queries to convert all installed tables to utf8mb4
+     *
+     * @return array
+     */
+    public static function getUtf8mb4ConversionQueries()
+    {
+        $allTables = DbHelper::getTablesInstalled();
+        $database  = Config::getInstance()->database['dbname'];
+
+        $queries   = [];
+        $queries[] ="ALTER DATABASE $database CHARACTER SET = utf8mb4 COLLATE = utf8mb4_unicode_ci;";
```

Comment:
Actually wondering if this one is really needed @sgiehl ? Eg thinking of WordPress or other applications where the same DB might be used for other applications too this might not be expected. I reckon it be good to remove this if possible. It could have many side effects when people use same DB for multiple apps which many do

Reply:
will remove it